in_toto/model.go

```diff
@@ -150,6 +190,67 @@ func (l *Layout) InspectAsInterfaceSlice() []interface{} {
 	return inspectionsI
 }
 
+func (l *Layout) validate() {
+	validateFunctions := []func(){l.validateType, l.validateExpires,
+		l.validateKeys, l.validateStepsAndInspections}
+	for _, f := range validateFunctions {
+		f()
+	}
+}
+
+func (l *Layout) validateType() {
+	if l.Type != "layout" {
+		fmt.Println("Invalid Type value for layout: should be 'layout'")
+	}
+}
+
+func (l *Layout) validateExpires() {
+	if _, err := time.Parse("2006-01-02T15:04:05Z", l.Expires); err != nil {
+		fmt.Println("Expiry time parsed incorrectly - date either invalid or" +
+			" of incorrect format")
+	}
+}
+
+func (l *Layout) validateKeys() {
+	for keyId, key := range l.Keys {
+
+		keyIdFormatCheck, err := regexp.MatchString("[a-fA-F0-9]+", keyId)
+		if err != nil {
+			fmt.Println("Unable to check if key ID has valid format")
+		}
+		if !keyIdFormatCheck {
+			fmt.Println("Key ID has invalid format")
+		}
+
+		if key.KeyId != keyId {
+			fmt.Println("Invalid key found")
+		}
+		// SSL Sets this to be optional. Can it ever be non empty?
```

Comment:
Yes and no. In a layout (i.e. here) the private portion of the key must be empty, but in general a `Key` can have public and private portions.